### DISCUSSION TITLE
[blocks]: Tag Input and File Upload

### DISCUSSION THREAD

**Discussion by colonel14** [shadcn-ui/ui#3178] [Blocks Request]
### Description

Can we pls have a block for Tag Input and file Upload

https://primereact.org/multiselect/

### Example

https://primereact.org/fileupload/

**Comment by MildTomato** [comment]
We we're thinking of making an [Uppy](https://uppy.io/) based _copy and paste_ Block for [supabase](https://github.com/supabase/supabase). We are looking at building out Supabase UI Blocks that use shadcn tailwind setup so anyone using shadcn can always copy and paste our blocks in easily.

Are there other libs maybe we should consider? Can't see an upload component in ui.shadcn yet, so I guess it's up for debate.

Any Block examples we do could always be submitted to shadcn site as well (minus the Supabase parts).

**Comment by birobirobiro** [comment]
File upload preview: https://file-vault-delta.vercel.app/

Code: https://github.com/ManishBisht777/file-vault

**Reply by c0h1b4** [reply]
GitHub button is not working.
Nice work, btw.

**Reply by birobirobiro** [reply]
@c0h1b4 Message updated 👊

**Comment by xiaoluoboding** [comment]
Tag input is a popular feature that provides an effective way to make multiple selections.

**Reply by trueberryless** [reply]
Like this: https://mantine.dev/core/tags-input/

**Comment by akarabach** [comment]
@sersavan is implemented multiselect, link to the repo https://github.com/sersavan/shadcn-multi-select-component/tree/main

**Comment by donovan-fintool** [comment]
It'd be great to get an official component for tag input / multi select. So many people have tried to reinventing the same thing (see https://github.com/shadcn-ui/ui/issues/3647).

**Comment by soranoo** [comment]
Uppy 5 is released and now they have headless component support, best fit for shadcn customisation

ref:
https://uppy.io/blog/uppy-5.0/
